doc/source/comparison_with_sas.rst

```diff
@@ -0,0 +1,604 @@
+shown in this document, almost any operation that can be applied to a data set
+using SAS's ``DATA`` step, can also be accomplished in pandas.
+
+A ``Series`` is the data structure that represents one column of a
+``DataFrame``. SAS doesn't have a separate data structure for a single column,
+but in general, working with a ``Series`` is analogous to referencing a column
+in the ``DATA`` step.
+
+``Index``
+~~~~~~~~~
+
+Every ``DataFrame`` and ``Series`` has an ``Index`` - which are labels on the
+*rows* of the data. SAS does not have an exactly analogous concept. A data set's
+row are essentially unlabeled, other than an implicit integer index that can be
+accessed during the ``DATA`` step (``_N_``).
+
+In pandas, if no index is specified, an integer index is also used by default
+(first row = 0, second row = 1, and so on). While using a labeled ``Index`` or
+``MultiIndex`` can enable sophisticated analyses and is ultimately an important
+part of pandas to understand, for this comparison we will essentially ignore the
+``Index`` and just treat the ``DataFrame`` as a collection of columns. Please
+see the :ref:`indexing documentation<indexing>` for much more on how to use an
+``Index`` effectively.
+
+
+Data Input / Output
+-------------------
+
+Reading keyed-in values:
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+A SAS data set can be built from specified values by
+placing the data after a ``datalines`` statement and
+specifying the column names.
+
+.. code-block:: none
+
+   data df;
+       input x y;
+       datalines;
+       1 2
+       3 4
+       5 6
+       ;
+   run;
+
+A pandas ``DataFrame`` can be constructed in many different ways,
+but for a small number of values, it is often convenient to specify it as
+a python dictionary, where the keys are the column names
+and the values are the data.
+
+.. ipython:: python
+
+   df = pd.DataFrame({
+             'x': [1, 3, 5],
+             'y': [2, 4, 6]})
+   df
+
+
+Reading external data:
+~~~~~~~~~~~~~~~~~~~~~~
+
+Like SAS, pandas provides utilities for reading in data from
+many formats.  The ``tips`` dataset, found within the pandas
+tests (`csv <https://raw.github.com/pydata/pandas/master/pandas/tests/data/tips.csv>`_)
+will be used in many of the following examples.
+
+SAS provides ``PROC IMPORT`` to read csv data into a data set.
+
+.. code-block:: none
+
+   proc import datafile='tips.csv' dbms=csv out=tips replace;
+       getnames=yes;
+   run;
+
+The pandas method is ``read_csv``, which works similarly.
+
+.. ipython:: python
+
+   url = 'https://raw.github.com/pydata/pandas/master/pandas/tests/data/tips.csv'
+   tips = pd.read_csv(url)
+   tips.head()
+
+
+Like ``PROC IMPORT``, ``read_csv`` can take a number of parameters to specify
+how the data should be parsed.  For example, if the data was instead tab delimited,
+and did not have column names, the pandas command would be:
+
+.. code-block:: python
+
+   tips = pd.read_csv('tips.csv', delim='\t', header=False)
```

Comment:
delim -> sep (delimiter is a valid keyword alternative to sep, but sep is the preferred one (listed first in the docstring))